safe-global/web-core
# Fix: error messages for non-owners

## What it solves

Resolves https://github.com/safe-global/web-core/issues/769, https://github.com/safe-global/web-core/issues/722

Contains #860 by @niklasnatter + my additions to handle the new tx case and gas limit errors.

## How this PR fixes it

Builds on top of @niklasnatter's PR and adjusts the owner condition:

```
const cannotPropose = !isOwner && !onlyExecute
```

Non-owners can only submit if execution is the only action. If a tx can be either signed or executed, it means it's not fully signed and thus cannot be executed by non-owners.

## How to test it
 It should show an error in the following cases:
* Create a tx and switch the wallet to a non-owner
* Start signing an existing tx and switch the wallet to a non-owner

It should NOT show an error when:
* Switch to a non-owner and try executing an existing fully signed tax

## Screenshots

<img width="575" alt="Screenshot 2022-10-08 at 08 36 26" src="https://user-images.githubusercontent.com/381895/194693835-dd93cb44-dcba-4f4c-b4ec-e099f08c58dc.png">
